pkg/controller/atlasproject/atlasproject_controller.go

```diff
@@ -59,7 +63,7 @@ func (r *AtlasProjectReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error
 		return reconcile.Result{}, nil
 	}
 	// This update will make sure the status is always updated in case of any errors or successful result
-	defer statushandler.Update(ctx, r, project)
+	defer statushandler.Update(ctx, r.Client, project)
```

Comment:
May I suggest we make `r.Client` a regular field as well? I bet there will be more method naming conflicts if we don't

Reply:
sure, I thought you've fixed this in your other PR?
I had to do this because for some reasons this stopped compiling - though I'm ok if your fix supersedes this

Reply:
My fix was for Cluster only to limit the scope of changes, which was large enough as it was.

We could, however, do this later as part of [CLOUDP-80607](https://jira.mongodb.org/browse/CLOUDP-80607) for the same reason.

Reply:
ok, I'll do this then!